exec/java-exec/src/main/java/org/apache/drill/exec/resourcemgr/rmblobmgr/RMConsistentBlobStoreManager.java

```diff
@@ -0,0 +1,354 @@
+    mapper.registerModule(deserModule);
+    return mapper;
+  }
+
+  private void initializeBlobs(Collection<QueryQueueConfig> leafQueues) throws Exception {
+    // acquire the global lock and ensure that all the blobs are created with empty data
+    int acquireTry = 1;
+    do {
+      try {
+        globalBlobMutex.acquire();
+        break;
+      } catch (Exception ex) {
+        ++acquireTry;
+      }
+    } while (acquireTry <= MAX_ACQUIRE_RETRY);
+
+    // if the lock is not acquired then just return as some other Drillbit can do it
+    // but there can be issues when none of the Drillbit is able to perform this operation
+    if (!globalBlobMutex.isAcquiredInThisProcess()) {
+      logger.warn("Failed to acquire global rm blobs lock to do blob initialization. Expectation is some other " +
+        "Drillbit should be able to do it");
+      return;
+    }
+
+    try {
+      logger.info("Acquired global rm blobs lock to do blob initialization");
+      // if here that means lock is acquired
+      rmStateBlobs.put(ClusterStateBlob.NAME,
+        new ClusterStateBlob(RM_STATE_BLOB_VERSION, new HashMap<>()));
+      rmStateBlobs.put(QueueLeadershipBlob.NAME,
+        new QueueLeadershipBlob(RM_STATE_BLOB_VERSION, new HashMap<>()));
+
+      // This ForemanResourceUsage blob needs to be per queue
+      final ForemanQueueUsageBlob queueUsageBlob = new ForemanQueueUsageBlob(RM_STATE_BLOB_VERSION, new HashMap<>());
+      for (QueryQueueConfig queueConfig : leafQueues) {
+        final String blobName = ForemanQueueUsageBlob.NAME + "_" + queueConfig.getQueueName();
+        rmStateBlobs.put(blobName, queueUsageBlob);
+      }
+
+      for (Map.Entry<String, RMStateBlob> stateBlob : rmStateBlobs.entrySet()) {
+        if (!rmBlobStore.putIfAbsent(stateBlob.getKey(), stateBlob.getValue())) {
+          logger.info("Blob {} was already initialized", stateBlob.getKey());
+        }
+      }
+    } catch (Exception ex) {
+      // consume the exception during blob initialization since we are expecting some other Drillbit can do that
+      // successfully. If not then there will be failure in cluster during actual blob update
+      logger.error("Failed to initialize one or more blob with empty data, but consuming this exception since " +
+        "expectation is that some other Drillbit should be able to perform this step");
+    } finally {
+      // throwing exception on release since it indicates mutex is in bad state
+      globalBlobMutex.release();
+    }
+  }
+
+  @Override
+  public void reserveResources(Map<String, NodeResources> queryResourceAssignment,
+                               QueryQueueConfig selectedQueue, String leaderId,
+                               String queryId, String foremanNode) throws Exception {
+    // Looks like leader hasn't changed yet so let's try to reserve the resources
+    // See if the call is to reserve or free up resources
+    Map<String, NodeResources> resourcesMap = queryResourceAssignment;
```

Comment:
Can this be changed to the following code.
    Map<String, NodeResources> resourcesMap =  queryResourceAssignment.entrySet().stream()
      .collect(Collectors.toMap(Map.Entry::getKey,
        (x) -> new NodeResources(x.getValue().getVersion(),
                                 -x.getValue().getMemoryInBytes(),
                                 -x.getValue().getNumVirtualCpu())));